src/briefcase/config.py

```diff
@@ -83,6 +84,70 @@ def make_class_name(formal_name):
     return class_name
 
 
+def validate_document_icon(document_type, document):
+    try:
+        if not isinstance(document["icon"], str):
+            raise BriefcaseConfigError(
+                f"An icon associated with a document type '{document_type}' is invalid."
```

Comment:
There's only one icon; and the error can be more explicit about what is wrong:
```suggestion
                f"The icon definition associated with document type {document_type!r} is not a string."
```